src/controllers/wordlist-controller.js

```diff
@@ -1,122 +1,223 @@
-import { PsEvent, LanguageModelFactory as LMF, Constants, Lemma, Inflection, DefinitionSet, Lexeme, Homonym } from 'alpheios-data-models'
+import { PsEvent } from 'alpheios-data-models'
 import WordList from '@/lib/word-list'
-import IndexedDBAdapter from '@/storage/indexed-db-adapter'
+import WordItem from '@/lib/word-item'
 
 export default class WordlistController {
-  constructor (userID) {
-    this.userID = userID
+  /**
+   * @constructor
+   * @param {String[]} availableLangs language codes
+   * @param {PSEvent[]} events events that the controller can subscribe to
+   */
+  constructor (availableLangs,events) {
     this.wordLists = {}
-
-    this.storageAdapter = new IndexedDBAdapter()
+    this.availableLangs = availableLangs
+    events.TEXT_QUOTE_SELECTOR_RECEIVED.sub(this.onTextQuoteSelectorReceived.bind(this))
+    events.HOMONYM_READY.sub(this.onHomonymReady.bind(this))
+    events.DEFS_READY.sub(this.onDefinitionsReady.bind(this))
+    events.LEMMA_TRANSL_READY.sub(this.onLemmaTranslationsReady.bind(this))
```

Comment:
I wanted to be able to decouple both the user data manager and the wordlist controller from the LexicalQuery, to keep the dependencies to a minimum. But I'm not happy with the way I did this. I don't really know what is the right way. I wanted the UIController to be able to say which things depend upon which, but it would require a fair amount of work to design something that allowed for that to be done fully. I think the User Data Manager and the Wordlist Controller would have to have to expose some API that allows for registering of events.  I'm not sure of the cost-benefit of premature optimization here.

Reply:
actually, after replaying our conversation from this morning in my head, I realized that @irina060981 's original approach to this, which was to have UIController subscribe the WordListController to the Lexical Query events, probably was closer to what I wanted than the change I ended up making.  So @irina060981 if you want to put that back, I agree that is better.  So sorry for confusing this more than necessary.
(i.e from UIController: 

```
LexicalQuery.evt.TEXT_QUOTE_SELECTOR_RECEIVED.sub(uiController.wordlistC.onTextQuoteSelectorRecieved.bind(uiController.wordlistC))	  
LexicalQuery.evt.LEXICAL_QUERY_COMPLETE.sub(uiController.wordlistC.onHomonymReady.bind(uiController.wordlistC))	    
LexicalQuery.evt.DEFS_READY.sub(uiController.wordlistC.onDefinitionsReady.bind(uiController.wordlistC))	
LexicalQuery.evt.LEMMA_TRANSL_READY.sub(uiController.wordlistC.onLemmaTranslationsReady.bind(uiController.wordlistC))
```